### ISSUE TITLE
[feature] ci + homebrew integration

### ISSUE THREAD

**Issue by cruzdanilo** [mapeditor/tiled#2849]
it would be nice if the travis ci script could create a new merge request automatically on [homebrew-cask](https://github.com/Homebrew/homebrew-cask) repository (or something like that) every new release.

**Comment by bjorn** [comment]
Definitely! Do you have an example of a project already doing this? Or could you help with setting that up?

It doesn't need to be triggered from Travis CI of course, it can also use GitHub Actions.

**Comment by PoonamMehan** [comment]
We could use, official Homebrew action (https://github.com/Homebrew/actions)
Maintained by the Homebrew team, so it's reliable long-term. It works by checking for new versions on a schedule (cron job), which means there's always a small delay between a Tiled release and the Homebrew update request being sent.

Another approach, eugenesvk's fork (https://github.com/eugenesvk/action-homebrew-bump-cask)
This one reacts immediately when a new release tag is published, making it faster and more direct. The downside is that it's maintained by a single person, which creates a dependency risk if the repo stops being actively maintained or patched over time.

Happy to hear thoughts on which direction makes more sense for the project, and can you assign me this issue? @bjorn 
> message from discord, pasting it here for tracking purposes

**Comment by bjorn** [comment]
@PoonamMehan I'll assign the issue to you, thank you for looking into this!

To me it sounds like eugenesvk will be the better option. We're not doing Tiled releases that often, so a cron job will always either check too frequently or take too long to update. Maybe it would even be best to set up this action as a manual trigger. It just needs to be easy but it can be part of the release process to trigger that job.

**Comment by PoonamMehan** [comment]
I have made a PR for the workflow file. On your end, you'll have to create a custom personal GitHub access token with the 'public_repo' and 'workflow' scopes. And add this token to repo's secrets as `HOMEBREW_BUMP_TOKEN`.

@bjorn
